injected/src/features/autofill-password-import.js

```diff
@@ -77,38 +102,164 @@ export default class AutofillPasswordImport extends ContentFeature {
                       style: this.signInButtonStyle,
                       element,
                       shouldTap: this.#signInButtonSettings?.shouldAutotap ?? false,
+                      shouldWatchForRemoval: false,
                   }
                 : null;
         } else {
             return null;
         }
     }
 
+    /**
+     * Removes the overlay if it exists.
+     */
+    removeOverlayIfNeeded() {
+        const existingOverlay = document.getElementById(OVERLAY_ID);
+        if (existingOverlay != null) {
+            existingOverlay.style.display = 'none';
+            existingOverlay.remove();
+            document.removeEventListener('scroll', this.updateOverlayPosition.bind(this));
+        }
+    }
+
+    /**
+     * Updates the position of the overlay based on the element to center on.
+     * @param {HTMLDivElement} overlay
+     * @param {any} style
+     */
+    updateOverlayPosition(overlay, style) {
+        const animations = overlay.getAnimations();
+        animations.forEach((animation) => animation.pause());
+        const { top, left, width, height } = this.elementToCenterOn.getBoundingClientRect();
+        overlay.style.position = 'absolute';
+
+        const isRound = style.borderRadius === '100%';
+
+        const widthOffset = isRound ? width / 2 : 0;
+        const heightOffset = isRound ? height / 2 : 0;
+
+        overlay.style.top = `calc(${top}px + ${window.scrollY}px - ${widthOffset}px - 1px - ${style.offsetTopEm}em)`;
+        overlay.style.left = `calc(${left}px + ${window.scrollX}px - ${heightOffset}px - 1px - ${style.offsetLeftEm}em)`;
+
+        // Ensure overlay is non-interactive
+        overlay.style.pointerEvents = 'none';
+        animations.forEach((animation) => animation.play());
+    }
+
+    /**
+     * Inserts an overlay element to animate, by adding a div to the body
+     * and styling it based on the found element.
+     * @param {HTMLElement|Element} mainElement
+     * @param {any} style
+     * @returns {HTMLElement|Element|null}
+     */
+    insertOverlayElement(mainElement, style) {
+        this.removeOverlayIfNeeded();
+
+        const overlay = document.createElement('div');
+        overlay.setAttribute('id', OVERLAY_ID);
+
+        if (this.elementToCenterOn != null) {
+            this.updateOverlayPosition(overlay, style);
+
+            const mainElementRect = mainElement.getBoundingClientRect();
+            overlay.style.width = `${mainElementRect.width}px`;
+            overlay.style.height = `${mainElementRect.height}px`;
+            overlay.style.zIndex = style.zIndex;
+
+            // Ensure overlay is non-interactive
+            overlay.style.pointerEvents = 'none';
+
+            document.addEventListener('scroll', () => {
```

Comment:
Should we use the `passive` argument here?

Reply:
Yep makes sense here!